src/zarr/core/metadata/v2.py

```diff
@@ -71,6 +71,14 @@ def __init__(
         shape_parsed = parse_shapelike(shape)
         dtype_parsed = parse_dtype(dtype)
         chunks_parsed = parse_shapelike(chunks)
+        if not filters and not compressor:
+            filters, compressor = _default_filters_and_compressor(dtype_parsed)
+        if dtype is str or dtype == "str":
+            vlen_codec: dict[str, JSON] = {"id": "vlen-utf8"}
+            if filters and not any(x["id"] == "vlen-utf8" for x in filters):
+                filters = list(filters) + [vlen_codec]
+            else:
+                filters = [vlen_codec]
```

Comment:
I'm wondering if this is really the right place for this. Or if it would be better higher up in the API (in the creation routines). My concern is that we may end up applying defaults to arrays that were intended to not use a filter/compressor.

Reply:
I agree this should be higher up. Either in the (Async)Array.create or even in the top-level zarr.open